corehq/apps/userreports/decorators.py

```diff
@@ -39,3 +41,32 @@ def _inner(*args, **kwargs):
         except TableNotFoundException:
             raise TableNotFoundWarning
     return _inner
+
+
+def ucr_context_cache(vary_on=()):
+    """
+    Decorator which caches calculations performed during a UCR EvaluationContext
+    The decorated function or method must have a parameter called 'context'
+    which will be used by this decorator to store the cache.
+    """
+    def decorator(fn):
+        assert 'context' in fn.func_code.co_varnames
+        assert isinstance(vary_on, (tuple, list))
```

Comment:
this could just be `tuple`  unless you see an instance where lists would be useful? I think we'd always want `vary_on` to be some constant?

Reply:
It gets coerced to a tuple below, so it doesn't really matter, as long as it's an iterable.  Are you thinking it'd be safer from mutation or something?

Reply:
I think of it as a code simplicity thing. I can't imagine ever not wanting a tuple here, so enforcing it seems reasonable. (tuples are slightly more performant as well, but that's likely not a real issue here)

Reply:
Yeah, I suppose it's no real burden on future implementers to require a tuple here.  I'll change it.